pool/store/memory/memory.go

```diff
@@ -275,10 +278,10 @@ func (s *memoryStore) UpdateNodePeers(nodeID store.NodeID, peers []string, block
 	now := time.Now()
 	node.LastSeen = now
 	node.BlockNumber = blockNumber
+
 	for _, peer := range peers {
 		// Only update peers we already know about
 		// FIXME: If symmetric peers disappear at the same time, then reappear, will it be a problem if they never become inactive? (Okay if the balance manager caps the update interval?)
-		// FIXME: Should the timestamp update happen on the peerNode also? Or is it okay to leave it for the symmetric update call?
```

Comment:
Turns out this question was the crux of the problem. :|

Banged my head at it for a while, then realized I was conflating the node's peer timestamps with the node's LastSeen in the badger implementation (whereas memory had this FIXME instead of actually doing it) which was breaking everything. Weeee!